src/main.rs

```diff
@@ -25,15 +52,45 @@ fn main() {
         .enable().unwrap();
     info!("Starting uoSQL server...");
 
+    // Getting the information for a possible configuration
+    let args : Args = Docopt::new(USAGE).and_then(|d| d.decode())
+                                        .unwrap_or_else(|e| e.exit());
+
+    // If a cfg is entered, use this file name to set configurations
+    let mut config = read_conf_from_json(args.flag_cfg
+                                .unwrap_or("src/config.json".into()));
+
+    // Change the IP address if flag is set
+    if let Some(s) = args.flag_ip {
+        config.address = string_to_ipv4(s);
+    }
+
+    // Change port if flag is set
+    if let Some(s) = args.flag_port {
+        config.port = s;
+    }
+
+    // Change directory is flag is set
+    if let Some(s) = args.flag_dir {
+        config.dir = s;
+    }
+
+    info!("IP: {}  Port: {}  Directory: {}",
+                        config.address, config.port, config.dir);
```

Comment:
Please change "IP" to "Bind-Address" or something. It's not really a "IP", but a mask that determines what connections to accept...